File: instrumentation/http4s-blaze-client-2.12_0.21/src/main/scala/com/nr/instrumentation/http4s/OutboundRequestWrapper.scala
package com.nr.instrumentation.http4s

import com.newrelic.api.agent.{HeaderType, OutboundHeaders}
import org.http4s.Request

class OutboundRequestWrapper[F[_]](val request: Request[F]) extends OutboundHeaders {
  /**
    * Return the type of header key syntax used for this.
    *
    * @return An <code>enum</code> specifying the type of headers present.
    * @since 3.5.0
    */
  override def getHeaderType: HeaderType = HeaderType.HTTP

  /**
    * Sets a response header with the given name and value.
    * NO-OP HTTP4s Request Headers are immutable and so can't be set from here
    */
  override def setHeader(name: String, value: String): Unit = {
  }
}

File: instrumentation/netty-reactor-http-1.0.0/src/main/java/com/nr/instrumentation/OutboundRequestWrapper.java
/*
 *
 *  * Copyright 2026 New Relic Corporation. All rights reserved.
 *  * SPDX-License-Identifier: Apache-2.0
 *
 */

package com.nr.instrumentation;

import com.newrelic.api.agent.HeaderType;
import com.newrelic.api.agent.OutboundHeaders;
import reactor.netty.http.client.HttpClientRequest;

public class OutboundRequestWrapper implements OutboundHeaders {

    private final HttpClientRequest request;

    public OutboundRequestWrapper(HttpClientRequest request) {
        this.request = request;
    }

    @Override
    public HeaderType getHeaderType() {
        return HeaderType.HTTP;
    }

    @Override
    public void setHeader(String name, String value) {
        request.addHeader(name, value);
    }
}